core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterContext.scala

```diff
@@ -0,0 +1,79 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.h2o.backends.external
+
+import org.apache.spark.h2o._
+import org.apache.spark.h2o.converters.WriteConverterContext
+import org.apache.spark.h2o.utils.NodeDesc
+import water.ExternalFrameWriterClient
+
+class ExternalWriteConverterContext(nodeDesc: NodeDesc) extends ExternalBackendUtils with WriteConverterContext {
+
+  val socketChannel = ConnectionToH2OHelper.getOrCreateConnection(nodeDesc)
+  val externalFrameWriter = new ExternalFrameWriterClient(socketChannel)
+
+  /**
+    * This method closes the communication after the chunks have been closed
+    */
+  override def closeChunks(): Unit = {
+    externalFrameWriter.waitUntilAllWriten()
+    ConnectionToH2OHelper.putAvailableConnection(nodeDesc, socketChannel)
```

Comment:
If that happens, connection is not released but will be removed by cleaning thread in `ConnectionHelper`. Do you think it's sufficient or have any other ideas ? 

btw: Thanks for typo!